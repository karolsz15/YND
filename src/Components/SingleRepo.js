import React from 'react';
import PropTypes from 'prop-types';
import classes from '../Container/Container.module.css';

const SingleRepo = (props) => {
  const {
    singleRepoLink,
    singleRepoTitle,
    singleRepoStars,
    singleRepoDescription,
  } = props;

  return (
    <a
      className={classes.Wrap}
      href={singleRepoLink}
      target="_blank"
      rel="noopener noreferrer"
    >
      <div className={classes.SingleRepo}>
        <div className={classes.TitleAndDescription}>
          <div className={classes.RepoTitle}>
            <div>{singleRepoTitle}</div>
            <div>
              {singleRepoStars}
              <i className="fa fa-star" />
            </div>
          </div>
          <div className={classes.RepoDescription}>{singleRepoDescription}</div>
        </div>
      </div>
    </a>
  );
};

SingleRepo.propTypes = {
  singleRepoLink: PropTypes.string.isRequired,
  singleRepoTitle: PropTypes.string.isRequired,
  singleRepoStars: PropTypes.number.isRequired,
  singleRepoDescription: PropTypes.string,
};

export default SingleRepo;
