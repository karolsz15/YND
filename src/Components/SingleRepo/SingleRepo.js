import React from 'react';
import PropTypes from 'prop-types';
import classes from '../../Container/Container.module.css';

const SingleRepo = ({
  singleRepoLink,
  singleRepoTitle,
  singleRepoStars,
  singleRepoDescription,
}) => (
  <a
    className={classes.Wrap}
    href={singleRepoLink}
    target="_blank"
    rel="noopener noreferrer"
  >
    <div className={classes.SingleRepo}>
      <div className={classes.TitleAndDescription}>
        <div className={classes.RepoTitle}>
          <div>{singleRepoTitle}</div>
          <div>
            {singleRepoStars}
            <i className="fa fa-star" />
          </div>
        </div>
        <div className={classes.RepoDescription}>{singleRepoDescription}</div>
      </div>
    </div>
  </a>
);

SingleRepo.propTypes = {
  singleRepoLink: PropTypes.string,
  singleRepoTitle: PropTypes.string,
  singleRepoStars: PropTypes.number,
  singleRepoDescription: PropTypes.string,
};

export default SingleRepo;
