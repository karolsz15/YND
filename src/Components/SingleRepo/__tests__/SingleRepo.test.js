import React from 'react';
import { shallow } from 'enzyme';
import SingleRepo from '../SingleRepo';

const wrapper = shallow(<SingleRepo />);

describe('SingleRepo component', () => {
  it('should render the SingleRepo correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});