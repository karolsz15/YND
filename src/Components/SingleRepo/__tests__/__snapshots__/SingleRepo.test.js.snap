// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRepo component should render the SingleRepo correctly 1`] = `
<a
  className="Wrap"
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="SingleRepo"
  >
    <div
      className="TitleAndDescription"
    >
      <div
        className="RepoTitle"
      >
        <div />
        <div>
          <i
            className="fa fa-star"
          />
        </div>
      </div>
      <div
        className="RepoDescription"
      />
    </div>
  </div>
</a>
`;
